<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Paulo Czarnewski" />

<meta name="date" content="2019-05-30" />

<title>Comparisson of DE genes in DSS colitis vs Human colitis</title>

<script src="mouse_colitis_vs_UC_files/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="mouse_colitis_vs_UC_files/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="mouse_colitis_vs_UC_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="mouse_colitis_vs_UC_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="mouse_colitis_vs_UC_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="mouse_colitis_vs_UC_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="mouse_colitis_vs_UC_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="mouse_colitis_vs_UC_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="mouse_colitis_vs_UC_files/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Comparisson of DE genes in DSS colitis vs Human colitis</h1>
<h3 class="subtitle">Czarnewski et al 2019</h3>
<h4 class="author">Paulo Czarnewski</h4>
<h4 class="date">30 May 2019</h4>

</div>


<div id="load-packages" class="section level1">
<h1><span class="header-section-number">1</span> Load Packages</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Installing required packages</span>
<span class="kw">source</span>(<span class="st">&quot;./support_functions.R&quot;</span>)
<span class="kw">inst_packages</span>(<span class="kw">c</span>(<span class="st">&quot;readxl&quot;</span>,<span class="st">&quot;data.table&quot;</span>,<span class="st">&quot;biomaRt&quot;</span>,<span class="st">&quot;RColorBrewer&quot;</span>,<span class="st">&quot;rafalib&quot;</span>,<span class="st">&quot;enrichR&quot;</span>,<span class="st">&quot;venn&quot;</span>))</code></pre></div>
<pre><code>##              Pkg_Status
## readxl             TRUE
## data.table         TRUE
## biomaRt            TRUE
## RColorBrewer       TRUE
## rafalib            TRUE
## enrichR            TRUE
## venn               TRUE</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Defining color paletes</span>
palete &lt;-<span class="st"> </span>RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>,<span class="st">&quot;Set2&quot;</span>)
mypalette2 &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">8</span>,<span class="dt">name =</span> <span class="st">&quot;Set2&quot;</span>),<span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">9</span>,<span class="dt">name =</span> <span class="st">&quot;Set1&quot;</span>))</code></pre></div>
<hr />
</div>
<div id="loading-human-dataset" class="section level1">
<h1><span class="header-section-number">2</span> Loading Human dataset</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">UC_DEgenes &lt;-<span class="st"> </span><span class="kw">data.frame</span>(readxl<span class="op">::</span><span class="kw">read_xlsx</span>(<span class="st">&quot;/Users/Czarnewski/Box Sync/Paulo_data/20170913 - Human UC treatment prediction/Treatment naive UC/jjx139_suppl_supplementary_table_1.xlsx&quot;</span>,<span class="dt">skip =</span> <span class="dv">1</span>))
<span class="kw">rownames</span>(UC_DEgenes) &lt;-<span class="st"> </span>UC_DEgenes[,<span class="dv">1</span>]
UC_DEgenes &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(UC_DEgenes[,<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>])</code></pre></div>
</div>
<div id="loading-mouse-human-homologs" class="section level1">
<h1><span class="header-section-number">3</span> Loading Mouse-human homologs</h1>
<p>Since we used the gene annotation from JAX in part of the anlaysis, it is likely that the list of differentially expressed genes might slightly vary depending on the updates in their database. For this reason, we also provided the list of genes obtained during our initial anlaysis in order to produce the exact same figures as in the paper.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#A list of homolog genes can be found in:</span>
<span class="co">#homologs &lt;- fread(&#39;http://www.informatics.jax.org/downloads/reports/HMD_HumanPhenotype.rpt&#39;)</span>
<span class="co">#homologs &lt;- data.frame(&quot;mouse&quot;=homologs$V5,&quot;human&quot;=homologs$V1)</span>

<span class="co">#Load the homolog gene list version downloaded in 2018-06-15 used in the paper.</span>
homologs &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;../processed_files/homologs.csv&quot;</span>,<span class="dt">row.names =</span> <span class="dv">1</span>)
<span class="kw">head</span>(homologs)</code></pre></div>
</div>
<div id="loading-mouse-dataset" class="section level1">
<h1><span class="header-section-number">4</span> Loading Mouse dataset</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#Loading mouse data</span>
mouse_DEgenes &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;../processed_files/DEgenes.csv&quot;</span>,<span class="dt">header =</span> T,<span class="dt">row.names =</span> <span class="dv">1</span>)
mouse_modules &lt;-<span class="st"> </span><span class="kw">read.csv</span>(<span class="st">&quot;../processed_files/gene_modules.csv&quot;</span>,<span class="dt">header =</span> T,<span class="dt">row.names =</span> <span class="dv">1</span>)</code></pre></div>
</div>
<div id="plotting-venn-diagram-for-genes-in-common" class="section level1">
<h1><span class="header-section-number">5</span> Plotting venn diagram for genes in common</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mouse_homologs &lt;-<span class="st"> </span><span class="kw">rownames</span>(mouse_DEgenes)
mouse_homologs &lt;-<span class="st"> </span>mouse_homologs[<span class="op">!</span>(<span class="kw">rownames</span>(mouse_DEgenes) <span class="op">%in%</span><span class="st"> </span>homologs[,<span class="dv">1</span>] )]
mouse_homologs &lt;-<span class="st"> </span><span class="kw">c</span>(mouse_homologs, <span class="kw">as.character</span>(homologs[homologs[,<span class="dv">1</span>] <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(mouse_DEgenes),<span class="dv">2</span>]))

<span class="kw">mypar</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">2</span>))
input  &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">human_DEG=</span><span class="kw">as.character</span>(<span class="kw">rownames</span>(UC_DEgenes)),
               <span class="dt">mouse_DEG=</span>mouse_homologs)
a &lt;-<span class="st"> </span><span class="kw">attr</span>(<span class="kw">venn</span>(input,<span class="dt">cexil =</span> <span class="dv">2</span>,<span class="dt">cexsn =</span> <span class="dv">2</span>), <span class="st">&quot;intersections&quot;</span>)
<span class="kw">title</span>(<span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Venn diagram for all DEGs&quot;</span>))</code></pre></div>
<p><img src="mouse_colitis_vs_UC_files/figure-html/unnamed-chunk-5-1.png" width="960" /></p>
</div>
<div id="plotting-venn-diagram-per-cluster" class="section level1">
<h1><span class="header-section-number">6</span> Plotting venn diagram per cluster</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">intersect_per_cluster &lt;-<span class="st"> </span><span class="kw">list</span>()

<span class="kw">mypar</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">2</span>))
<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">max</span>(mouse_modules[,<span class="dv">1</span>])){
  mouse_homologs_module &lt;-<span class="st"> </span><span class="kw">rownames</span>(mouse_DEgenes)[mouse_modules[,<span class="dv">1</span>] <span class="op">==</span><span class="st"> </span>i]
  mouse_homologs_module &lt;-<span class="st"> </span>mouse_homologs_module[<span class="op">!</span>(<span class="kw">rownames</span>(mouse_DEgenes)[mouse_modules[,<span class="dv">1</span>] <span class="op">==</span><span class="st"> </span>i] <span class="op">%in%</span><span class="st"> </span>homologs[,<span class="dv">1</span>] )]
  mouse_homologs_module &lt;-<span class="st"> </span><span class="kw">c</span>(mouse_homologs_module, <span class="kw">as.character</span>(homologs[homologs[,<span class="dv">1</span>] <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(mouse_DEgenes)[mouse_modules[,<span class="dv">1</span>] <span class="op">==</span><span class="st"> </span>i],<span class="dv">2</span>]))
  
  input  &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">human_DEG=</span><span class="kw">as.character</span>(<span class="kw">rownames</span>(UC_DEgenes)),
                 <span class="dt">mouse_DEG=</span>mouse_homologs_module)
  a&lt;-<span class="kw">attr</span>(<span class="kw">venn</span>(input,<span class="dt">cexil =</span> <span class="fl">1.5</span>,<span class="dt">cexsn =</span> <span class="fl">1.5</span>), <span class="st">&quot;intersections&quot;</span>)
  <span class="kw">title</span>(<span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Module &quot;</span>,i))
  
  intersect_per_cluster[[i]] &lt;-<span class="st"> </span>a<span class="op">$</span><span class="st">`</span><span class="dt">human_DEG:mouse_DEG</span><span class="st">`</span>
}</code></pre></div>
<p><img src="mouse_colitis_vs_UC_files/figure-html/unnamed-chunk-6-1.png" width="960" /></p>
</div>
<div id="plotting-percentages-per-cluster" class="section level1">
<h1><span class="header-section-number">7</span> Plotting percentages per cluster</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">temp &lt;-<span class="st"> </span><span class="kw">sapply</span>(intersect_per_cluster,length)
<span class="kw">names</span>(temp) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;module&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="kw">max</span>(mouse_modules[,<span class="dv">1</span>]))

<span class="kw">mypar</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">2</span>,<span class="dv">3</span>))
<span class="kw">barplot</span>(temp[<span class="kw">order</span>(temp)],<span class="dt">horiz=</span>T,<span class="dt">yaxs=</span><span class="st">&quot;i&quot;</span>,<span class="dt">col=</span>mypalette2[<span class="dv">1</span><span class="op">:</span><span class="kw">max</span>(mouse_modules[,<span class="dv">1</span>])][<span class="kw">order</span>(temp)],<span class="dt">las=</span><span class="dv">1</span>,<span class="dt">border=</span><span class="ot">NA</span>,<span class="dt">xaxs=</span><span class="st">&quot;i&quot;</span>,<span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">200</span>),<span class="dt">xlab=</span><span class="st">&#39;number of genes&#39;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">0</span>,<span class="dt">lwd=</span><span class="dv">2</span>)
<span class="kw">text</span>(temp[<span class="kw">order</span>(temp)],(<span class="dv">0</span><span class="op">:</span>(<span class="kw">length</span>(temp)<span class="op">-</span><span class="dv">1</span>)<span class="op">*</span><span class="fl">1.2</span>)<span class="op">+</span><span class="fl">0.7</span>,<span class="kw">paste0</span>(<span class="dv">100</span><span class="op">*</span><span class="kw">round</span>(temp[<span class="kw">order</span>(temp)]<span class="op">/</span><span class="kw">sum</span>(temp),<span class="dv">3</span>),<span class="st">&quot;%&quot;</span>),<span class="dt">pos=</span><span class="dv">4</span>)
<span class="kw">pie</span>(temp[<span class="kw">order</span>(temp,<span class="dt">decreasing =</span> T)],<span class="dt">col =</span> mypalette2[<span class="dv">1</span><span class="op">:</span><span class="kw">max</span>(mouse_modules[,<span class="dv">1</span>])][<span class="kw">order</span>(temp,<span class="dt">decreasing =</span> T)],<span class="dt">clockwise =</span> T,<span class="dt">border =</span> <span class="st">&quot;white&quot;</span>,<span class="dt">main =</span> <span class="st">&quot;Percentage of conserved gene per cluster&quot;</span>)</code></pre></div>
<p><img src="mouse_colitis_vs_UC_files/figure-html/unnamed-chunk-7-1.png" width="960" /></p>
</div>
<div id="performing-gene-set-analysis-on-the-shared-list-between-mouse-and-humans" class="section level1">
<h1><span class="header-section-number">8</span> Performing Gene Set Analysis on the shared list between mouse and humans</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pvalue_cutoff &lt;-<span class="st"> </span><span class="fl">0.05</span>
no_genes_cutoff &lt;-<span class="st"> </span><span class="dv">3</span>


<span class="co">#GO enrichment</span>
a &lt;-<span class="st"> </span><span class="kw">enrichr</span>(<span class="dt">genes=</span><span class="kw">unique</span>(<span class="kw">unlist</span>(intersect_per_cluster)), <span class="dt">databases =</span> <span class="st">&quot;GO_Biological_Process_2017&quot;</span>)[[<span class="dv">1</span>]]
  <span class="co">#remove terms with p-values &gt; pvalue_cutoff and number of genes &lt; no_genes_cutoff</span>
a &lt;-<span class="st"> </span>a[<span class="kw">order</span>(a<span class="op">$</span>P.value,<span class="dt">decreasing =</span> F),]
a &lt;-<span class="st"> </span>a[a<span class="op">$</span>P.value <span class="op">&lt;</span><span class="st"> </span>pvalue_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(a[,<span class="st">&quot;Overlap&quot;</span>],<span class="st">&quot;/&quot;</span>),<span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>)) <span class="op">&gt;=</span><span class="st"> </span>no_genes_cutoff,]
  <span class="co">#remove terms with &quot;positive/negative regulation&quot;</span>
a &lt;-<span class="st"> </span>a[<span class="kw">grep</span>(<span class="st">&quot;regulation&quot;</span>,a<span class="op">$</span>Term,<span class="dt">invert =</span> T),]
  <span class="co">#if (nrow(a) &gt; 20) { a &lt;- a[order(a$P.value,decreasing = F),][1:20,] }</span>
a<span class="op">$</span>Term &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(a[,<span class="st">&quot;Term&quot;</span>],<span class="st">&quot; [(]GO:&quot;</span>),<span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>)


<span class="co">#KEGG enrichment</span>
b &lt;-<span class="st"> </span><span class="kw">enrichr</span>(<span class="dt">genes=</span><span class="kw">unique</span>(<span class="kw">unlist</span>(intersect_per_cluster)), <span class="dt">databases =</span> <span class="st">&quot;KEGG_2016&quot;</span>)[[<span class="dv">1</span>]]
b &lt;-<span class="st"> </span>b[<span class="kw">order</span>(b<span class="op">$</span>P.value,<span class="dt">decreasing =</span> F),]
b &lt;-<span class="st"> </span>b[<span class="kw">grep</span>(<span class="st">&quot;_hsa05&quot;</span>,b<span class="op">$</span>Term,<span class="dt">invert =</span> T),]   <span class="co">#Remove disease-related KEGG pathways (&quot;hsa05&quot;&quot;)</span>
b &lt;-<span class="st"> </span>b[b<span class="op">$</span>P.value <span class="op">&lt;</span><span class="st"> </span>pvalue_cutoff <span class="op">&amp;</span><span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sapply</span>(<span class="kw">strsplit</span>(b[,<span class="st">&quot;Overlap&quot;</span>],<span class="st">&quot;/&quot;</span>),<span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>)) <span class="op">&gt;=</span><span class="st"> </span>no_genes_cutoff,]
  <span class="co">#if (nrow(b) &gt; 20) { b &lt;- b[order(b$P.value,decreasing = F),][1:20,] }</span>
b<span class="op">$</span>Term &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(b[,<span class="st">&quot;Term&quot;</span>],<span class="st">&quot;_&quot;</span>),<span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>)


<span class="co">#Plot enrichments</span>
n &lt;-<span class="st"> </span><span class="dv">10</span>
<span class="kw">mypar</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">15</span>,<span class="dv">2</span>,<span class="dv">3</span>))
<span class="kw">barplot</span>(<span class="op">-</span><span class="kw">log</span>(a[n<span class="op">:</span><span class="dv">1</span>,<span class="st">&quot;P.value&quot;</span>],<span class="dv">10</span>),<span class="dt">names.arg=</span>a[n<span class="op">:</span><span class="dv">1</span>,<span class="dv">1</span>],<span class="dt">horiz=</span>T,<span class="dt">las=</span><span class="dv">1</span>,<span class="dt">col=</span><span class="st">&quot;grey&quot;</span>,<span class="dt">main=</span><span class="st">&#39;GO&#39;</span>,<span class="dt">xlab=</span><span class="st">&quot; &quot;</span>,<span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="kw">max</span>(<span class="op">-</span><span class="kw">log</span>(a[n<span class="op">:</span><span class="dv">1</span>,<span class="st">&quot;P.value&quot;</span>],<span class="dv">10</span>))<span class="op">*</span><span class="fl">1.2</span>),<span class="dt">tck=</span><span class="op">-</span>.<span class="dv">05</span>,<span class="dt">border=</span><span class="ot">NA</span>,<span class="dt">cex.axis=</span><span class="dv">1</span>,<span class="dt">cex.names=</span><span class="dv">1</span>)
<span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">0</span>,<span class="dt">lwd=</span><span class="dv">2</span>)
<span class="kw">barplot</span>(<span class="op">-</span><span class="kw">log</span>(b[n<span class="op">:</span><span class="dv">1</span>,<span class="st">&quot;P.value&quot;</span>],<span class="dv">10</span>),<span class="dt">names.arg=</span>b[n<span class="op">:</span><span class="dv">1</span>,<span class="dv">1</span>],<span class="dt">horiz=</span>T,<span class="dt">las=</span><span class="dv">1</span>,<span class="dt">col=</span><span class="st">&quot;grey&quot;</span>,<span class="dt">main=</span><span class="st">&#39;KEGG&#39;</span>,<span class="dt">xlab=</span><span class="st">&quot; &quot;</span>,<span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="kw">max</span>(<span class="op">-</span><span class="kw">log</span>(b[n<span class="op">:</span><span class="dv">1</span>,<span class="st">&quot;P.value&quot;</span>],<span class="dv">10</span>))<span class="op">*</span><span class="fl">1.2</span>),<span class="dt">tck=</span><span class="op">-</span>.<span class="dv">05</span>,<span class="dt">border=</span><span class="ot">NA</span>,<span class="dt">cex.axis=</span><span class="dv">1</span>,<span class="dt">cex.names=</span><span class="dv">1</span>)
<span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">0</span>,<span class="dt">lwd=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="mouse_colitis_vs_UC_files/figure-html/unnamed-chunk-8-1.png" width="960" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
